shared-bindings/socketpool/SocketPool.c

```diff
@@ -123,7 +123,7 @@ STATIC mp_obj_t socketpool_socketpool_getaddrinfo(size_t n_args, const mp_obj_t
     }
 
     if (ip_str == mp_const_none) {
-        mp_raise_OSError(0);
+        mp_raise_OSError(-2);
```

Comment:
It seems like the value passed to `mp_raise_OSError` is generally a positive errno value.  
```
py/mperrno.h:#define MP_ENOENT             (2) // No such file or directory
...
ports/unix/modffi.c:        mp_raise_OSError(MP_ENOENT);
```
but sometimes they're raised as negative values
```
extmod/vfs_lfsx.c:            mp_raise_OSError(-MP_ENOENT);
```

Does the positive/negative distinction mean something?  Should a named constant such as MP_NOENT be used here instead of the number 2?

Reply:
This weird because in CPython it is this error: https://docs.python.org/3/library/socket.html#socket.gaierror The value comes from:
```
Python 3.9.6 (default, Jun 30 2021, 10:22:16) 
[GCC 11.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import socket
>>> socket.EAI_NONAME
-2
```

I've added a short comment about what the value is.